docs/metrics/pearson-correlation.md

```diff
@@ -0,0 +1,56 @@
+---
+description: How to calculate the strength of linear correlation between 2 variables
+---
+
+# Pearson Correlation
+
+Pearson Correlation is a statistical measure that evaluates the strength and direction of the linear
+relationship between two continuous variables. The values range from -1 to +1. A value close to +1 indicates a strong
+positive linear correlation, where as one variable increases, the other also increases proportionally. A value close
+to -1 signifies a strong negative linear correlation, meaning as one variable increases, the other decreases
+proportionally. A value of 0 indicates no linear correlation between the variables.
+
+## Implementation Details
+
+The correlation coefficient is calculated by dividing the covariance of $x$ and $y$ by their individual standard
+deviations. This can be mathematically represented as:
+
+$$
+r = \frac{\text{cov}(x, y)}{\sigma_x \sigma_y} = \frac{\sum_{i=1}^{N} (x_i - \bar{x})(y_i - \bar{y})}{\sqrt{\sum_{i=1}^{N}
+(x_i - \bar{x})^2 \sum_{i=1}^{N} (y_i - \bar{y})^2}}
+$$
+
+where $x$ and $y$ are the collection of data points.
+
+### Examples
+
+Temperature Correlation:
+
+<div class="grid" markdown>
+
+| Ground Truth Temperature (&deg;C) | Predicted Temperature (&deg;C) |
+| --- | --- |
+| 25  | 30  |
+| 35  | 28  |
+| 30  | 27  |
+
+</div>
+
+$$
+\begin{align}
+\bar{x} &= \frac{25 + 35 + 30}{3} = 30 \\[1em]
+\bar{y} &= \frac{30 + 28 + 27}{3} = 28.33 \\[1em]
+r &= \frac{\sum_{i=1}^{4} (x_i - \bar{x})(y_i - \bar{y})}{\sqrt{\sum_{i=1}^{3} (x_i - \bar{x})^2 \sum_{i=1}^{3} (y_i -
+\bar{y})^2}} \\[1em]
+&= \frac{(25 - 30)(30 - 28.33) + (35 - 30)(28 - 28.33) + (30 - 30)(27 - 28.33)}{\sqrt{[(25 - 30)^2 +
+(35 - 30)^2 + (30 - 30)^2] [(30 - 28.33)^2 + (28 - 28.33)^2 + (27 - 28.33)^2 ]}} \\[1em]
+&\approx -0.65
```

Comment:
Should we use an example with a better model, i.e. one that produces a positive correlation? Same for Spearman's

Reply:
Oh yeah, I was also thinking about this - I think temperatures vs. predicted temperatures is not the best example, since correlation metrics are used to measure correlation between 2 variables, not necessarily predicted vs ground truth (we expect the correlation to always be positive so it's not so meaningful). For example, we might calculate the correlation between the person's IQ and their height to see if there is correlation. But I couldn't think of any realistic example with relatively small numbers. Maybe I could rename the columns to "Temperature in place A" and "Temperature in place B"?

Reply:
I think what @gordonhart meant here is to update the predicted temperature so it will yield a positive correlation. Regarding the column renaming, I prefer to keep it "temperatures vs. predicted temperatures" as this encourages people to use this correlation coefficient as an evaluation metric.

Reply:
Sounds good, I changed it now